detector_test.go

```diff
@@ -959,39 +965,22 @@ func testDetector(t *testing.T, when spec.G, it spec.S) {
 					Buildpack: buildpack.Info{ID: "B", Version: "v1"},
 				})
 
+				// try resolve
 				fifthGroup := []buildpack.GroupElement{
 					{ID: "D", Version: "v1", API: "0.9", Extension: true, Optional: true},
 					{ID: "B", Version: "v1", API: "0.8"},
 				}
-				fifthResolve := resolver.EXPECT().Resolve(
-					fifthGroup,
-					detector.Runs,
-				).Return(
-					[]buildpack.GroupElement{},
-					[]platform.BuildPlanEntry{},
-					lifecycle.ErrFailedDetection,
```

Comment:
Instead of forcing this group to fail we let it succeed in order to test the output when there are extensions.